### DISCUSSION TITLE
xDrip+ with Material You icon?

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2539] [Ideas]
Dear developers,
could you may add the configuration for a material you based icon for xDrip Plus?
I updated to Android 13 a few hours ago on my Find X5 Pro, and I have the "Background-Based" icons now, so that I'm able to customize my Icons. Please add the function also to xDrip plus!
I will add a screenshot to show you exactly what I mean.

Best regards!

**Comment by Miquest** [comment]
I would be very happy if you would implement an icon like the others, which depend on my wallpaper!

![Screenshot_2022-12-08-15-32-17-17_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/91971937/206473239-2eac91b1-e6a3-4759-8b52-ed680d4226df.jpg)

**Reply by Miquest** [reply]
I read on Stackoverflow, there is a (in my opinion) simple way to add a support for material you for Android Apps?

**Reply by emp-00** [reply]
What exactly is the problem? What exactly do you want to change? Can you describe explicitly what you mean, maybe show a screenshot of the changed/optimized "icon" or whatever you mean? Sorry, I cannot follow without more information.

**Reply by Miquest** [reply]
Okay, I'll explain it in detail again. Sorry, yesterday when I wrote the post I was in a hurry 😅
Okay, do you know what Material You is? 
Google implemented it within Android 12, so it's available since 1 year. It makes all icons unified, everything in the same color, based on the Wallpaper you use.
I'll add a screenshot in a minute. 
My question is, if it is possible to implement Material You Design for the xDrip Plus icon?
With a little search on the internet I found instructions, the link comes with the screenshot. You would have to add the xDrip Plus icon in monchrome and you would have to edit a configuration file, I think one of the .xml files. 

That would make that the xDrip icon can joyn the unified icons on my and also on other phones.

Kind regards,
Thorin E.

**Reply by Miquest** [reply]
With Material Design:

![Screenshot_2022-12-09-07-18-19-42_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/91971937/206637664-918e7156-1a79-48bb-8b05-62b928ab3f50.jpg)

And without it:
![Screenshot_2022-12-09-07-18-36-82_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/91971937/206637670-66e81a2b-8b7d-4112-9e90-989fab452be5.jpg)

And here is the link for the description how to implement my idea, if you want to do that:
https://stackoverflow.com/questions/71109063/how-to-implement-a-themed-monochrome-icon-for-material-you

**Reply by Miquest** [reply]
Google implemented the Feature for 3rd Party Apps within Android 13, so it wasn't available before.

**Reply by emp-00** [reply]
Aah, thanks. Now I understand, sorry I'm on Android 12 and thus had no idea... However, this is a pure "optical gimmick" from my perspective and personally I'd probably switch off "Material You" asap once I'd have Android 13, really not my taste :-)

Regarding the changes to xDrip, assuming this does not destroy the look-and-feel for Android 12 and below: If you have the solution at hand (see your link) and if its compatible with old Android's - feel free to make the small adjustments and post it as a pull request on github.

**Reply by Miquest** [reply]
Yes, it's a gimmick... :-)
I'll see what I can do to implement it, and I'll also test it at other Android phones with Android 12.
